rclcpp/include/rclcpp/wait_result.hpp

```diff
@@ -276,11 +276,11 @@ class WaitResult final
       auto & wait_set = this->get_wait_set();
       auto rcl_wait_set = wait_set.get_rcl_wait_set();
       while (next_waitable_index_ < wait_set.size_of_waitables()) {
-        auto cur_waitable = wait_set.waitables(next_waitable_index_);
+        auto cur_waitable = wait_set.waitables(next_waitable_index_++);
```

Comment:
I personally consider these constructs bad style. I always need to think twice, if the returned value of x++ is the pre or post incremented value, and find code written this way therefore hard to read.

Reply:
Ok, I can rework that to not use the postfix operator, but it was the smallest change I could make to fix the problem (not iterate after `break`). I may do that in a small follow pr though if we decide to merge this, because I want to get some soak time on this change before the release.

Reply:
I will do this in a follow up so we can merge.